whylog/config/abstract_config.py

```diff
@@ -222,49 +222,72 @@ def _get_search_ranges(cls, suspected_rules, effect_clues):
-                    }
-                },
-                'database': {
-                    'date': {
-                        InvesitgationStep.LEFT_BOUND: datetime(2016, 5, 29, 12, 32, 0),
-                        InvesitgationStep.RIGHT_BOUND: datetime(2016, 5, 29, 12, 33, 20)
-                    }
+        Sample rule's search ranges (1) : {
+            'apache': {
+                'date': {
+                    InvesitgationStep.LEFT_BOUND: datetime(2016, 5, 29, 12, 33, 0),
+                    InvesitgationStep.RIGHT_BOUND: datetime(2016, 5, 29, 12, 33, 30)
+                }
+            },
+            'database': {
+                'date': {
+                    InvesitgationStep.LEFT_BOUND: datetime(2016, 5, 29, 12, 32, 0),
+                    InvesitgationStep.RIGHT_BOUND: datetime(2016, 5, 29, 12, 33, 20)
+                }
+            }
+         }
+        Sample rule's search ranges (2) : {
+            'apache': {
+                'date': {
+                    InvesitgationStep.LEFT_BOUND: datetime(2016, 5, 29, 12, 32, 0),
+                    InvesitgationStep.RIGHT_BOUND: datetime(2016, 5, 29, 12, 33, 20)
                 }
-             }
+            },
+        }
         This method sums all rule's search ranges from every rule in suspected rules.
+        Expected returned value based on (1) and (2): {
+            'apache': {
+                'date': {
+                    InvesitgationStep.LEFT_BOUND: datetime(2016, 5, 29, 12, 32, 0),
+                    InvesitgationStep.RIGHT_BOUND: datetime(2016, 5, 29, 12, 33, 30)
+                }
+            },
+            'database': {
+                'date': {
+                    InvesitgationStep.LEFT_BOUND: datetime(2016, 5, 29, 12, 32, 0),
+                    InvesitgationStep.RIGHT_BOUND: datetime(2016, 5, 29, 12, 33, 20)
+                }
+            }
+         }
         """
         # This implementation assumes that all primary key groups is a one element list
         # TODO implementation for longer primary key groups
         search_ranges = {}
         for rule in suspected_rules:
             rule_search_ranges = rule.get_search_ranges(effect_clues)
             for log_type_name, log_type_ranges in six.iteritems(rule_search_ranges):
-                if log_type_name not in search_ranges:
+                log_type_search_range = search_ranges.get(log_type_name)
+                if log_type_search_range is None:
                     search_ranges[log_type_name] = log_type_ranges
                     continue
-                log_type_search_range = search_ranges[log_type_name]
-                for key_type, type_range in six.iteritems(rule_search_ranges[log_type_name]):
-                    if key_type not in log_type_search_range:
-                        log_type_search_range[key_type] = type_range
+                for type, range in six.iteritems(rule_search_ranges[log_type_name]):
```

Comment:
`type` and `range` are python keywords/builtins. So please replace with `type_` and `range_`